### PR TITLE
[breaking] Ignore infinite variable bounds

### PR DESCRIPTION
**Pull request by odow**
Introducing non-finite variable bounds is just a cause of trouble: https://github.com/jump-dev/MathOptInterface.jl/issues/1270

Setting a bound to `-Inf` or `Inf` after creation is just a sign that the user wants to delete the bound. So we should make them do it explicitly.

This is breaking because people may be relying on `has_lower_bound` returning `true` if the bound is `-Inf`.

Closes https://github.com/jump-dev/MathOptInterface.jl/issues/1270

### CODE REVIEW

File: src/variables.jl [hunk 1]
```diff
@@ -109,6 +109,42 @@ struct VariableInfo{S,T,U,V}
...
+        has_ub::Bool,
+        upper_bound::T,
+        has_fix::Bool,
+        fixed_value::U,
+        has_start::Bool,
+        start::V,
+        binary::Bool,
+        integer::Bool,
+    ) where {S,T,U,V}
+        if has_lb && lower_bound === -Inf
```
[the remark this answers is not in the document]
Comment (odow) [reply src/variables.jl]: Swapped for `isfinite`, which also checks for +/- inf and NaNs.

**Comment by odow** [comment]
Bump. @blegat how's this?

**Review by blegat — APPROVED** [review]
Since JuMP uses MOI.LessThan and MOI.GreaterThan (instead of MOI.Interval), it indeed does should not accept infinite bounds. We should wait before v0.22 is approaching before merging though.

File: src/macros.jl [hunk 2]
```diff
@@ -766,12 +766,12 @@ that either `func` or `set` will be some custom type, rather than e.g. a
 `Symbol`, since we will likely want to dispatch on the type of the function or
 set appearing in the constraint.
 
-For extensions that need to create constraints with more information than just 
-`func` and `set`, an additional positional argument can be specified to 
-`@constraint` that will then be passed on `build_constraint`. Hence, we can 
-enable this syntax by defining extensions of 
-`build_constraint(_error, func, set, my_arg; kw_args...)`. This produces the 
-user syntax: `@constraint(model, ref[...], expr, my_arg, kw_args...)`. 
+For extensions that need to create constraints with more information than just
```
Comment (blegat) [inline src/macros.jl:769]: Do this in a separate PR, this does not need to wait until v0.22

Comment (odow) [reply src/macros.jl:769]: oops. I think this just got caught up in the formatting.
